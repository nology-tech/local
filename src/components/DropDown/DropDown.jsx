import "./DropDown.scss";
import triangle from "../../assets/icons/triangleUpsideDown.svg";
import { useState } from "react";

const DropDown = ({ options = [], onChange }) => {
  const [dropDownActive, setDropDownActive] = useState(false);
  const [activeLocation, setActiveLocation] = useState("All");

  const handleDropDownActive = () => {
    setDropDownActive(!dropDownActive);
  };

  const handleActiveLocation = (location) => {
    onChange(location);
    setActiveLocation(`${location}`);
  };

  const applyActiveClassDiv = dropDownActive
    ? "dropdown__item dropdown__item--primary dropdown__item--primary--active"
    : "dropdown__item dropdown__item--primary";

  const applyActiveClassArrow = dropDownActive
    ? "dropdown__arrow arrow-active"
    : "dropdown__arrow";

  return (
    <div className="dropdown">
      <div
        role="select"
        tabIndex={0}
        className={applyActiveClassDiv}
        onClick={handleDropDownActive}
      >
        <p className="dropdown__item-name dropdown__item-name--primary">
          {activeLocation}
        </p>
        <div className={"dropdown__image-container"}>
          <img
            className={applyActiveClassArrow}
            src={triangle}
            alt="triangle"
          />
        </div>
      </div>
      <div className="dropdown__options">
        {dropDownActive &&
          options.map((location) => (
            <div
              key={location}
              role="option"
              tabIndex={0}
              className="dropdown__item dropdown__item--secondary"
              onClick={() => {
                handleActiveLocation(location);
                handleDropDownActive(false);
              }}
            >
              <p className="dropdown__item-name">{location}</p>
            </div>
          ))}
      </div>
    </div>
  );
};

export default DropDown;
