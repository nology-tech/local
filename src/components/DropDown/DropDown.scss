@use "../../../src/styles/variables" as *;
@use "../../styles/mixins" as *;

.dropdown__content {
  .dropdown__link--secondary:last-child {
    border-bottom-left-radius: 2rem;
    border-bottom-right-radius: 2rem;
  }
}

.dropdown {
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-evenly;
  align-items: center;

  padding: 1rem;

  .dropdown__item {
    background-color: $white;
    min-width: 15rem;
    display: flex;
    flex-flow: row;
    justify-content: space-between;
    align-items: center;
    border: 2px solid $black;
    padding: 1rem;
    &--primary {
      text-align: center;
      border-radius: 2rem;
      &--active {
        border-bottom-left-radius: 0;
        border-bottom-right-radius: 0;
        text-align: left;
      }
    }
    &:last-child {
      border-bottom-left-radius: 2rem;
      border-bottom-right-radius: 2rem;
    }
    .dropdown__item-name--primary {
      width: 60%;
    }

    .dropdown__item--secondary {
      position: relative;
    }
    .dropdown__image-container {
      display: block;
      width: 20%;
      height: 50%;
      justify-self: flex-end;
    }
  }
  .dropdown__arrow {
    width: 100%;
    height: 100%;
  }
}

.arrow-active {
  transform: rotate(180deg);
}
