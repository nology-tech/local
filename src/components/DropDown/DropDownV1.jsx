const [dropDownActive, setDropDownActive] = useState(false);
let dropDownClass = "dropdown__link dropdown__link--primary";
("dropdown__link dropdown__link--primary menu-active");
const handleDropDownActive = () => {
  console.log("hi");
  setDropDownActive(!dropDownActive);
  if (dropDownActive) {
    dropDownClass += " menu-active";
  } else {
    dropDownClass = "dropdown__link dropdown__link--primary";
  }
  return dropDownClass;
};
