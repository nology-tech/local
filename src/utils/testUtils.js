import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
