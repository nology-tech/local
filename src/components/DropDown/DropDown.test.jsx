import DropDown from "./DropDown";
import { customRender } from "../../utils/testUtils";
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

it("should render the dropdown", () => {
  const { container } = customRender(<DropDown />);

  expect(container).toMatchSnapshot();
});

it("should call the function when user selects an option from the dropdown menu", () => {
  const mockFunction = jest.fn();
  const mockData = ["London", "Newcastle"];
  customRender(<DropDown onChange={mockFunction} options={mockData} />);

  const dropDown = screen.getByRole("select");

  userEvent.click(dropDown);

  const dropDownOption = screen.getByText("London");

  userEvent.click(dropDownOption);

  expect(mockFunction).toBeCalled();
});
