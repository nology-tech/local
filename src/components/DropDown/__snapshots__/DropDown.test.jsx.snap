// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the dropdown 1`] = `
<div>
  <div
    class="dropdown"
  >
    <div
      class="dropdown__item dropdown__item--primary"
      role="select"
      tabindex="0"
    >
      <p
        class="dropdown__item-name dropdown__item-name--primary"
      >
        All
      </p>
      <div
        class="dropdown__image-container"
      >
        <img
          alt="triangle"
          class="dropdown__arrow"
          src="triangleUpsideDown.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
